### ISSUE TITLE
Update SelectPanel guidelines: Add clearing guidelines for SelectPanel/SelectMenu with single/multi selections

### ISSUE THREAD

**Issue by maximedegreve** [primer/design#229]
Currently our components are lacking guidelines as well as functionality on how clearing selections in a SelectPanel or SelectMenu should work. This is especially a problem when you have multiple items selected and quickly want to clear those

Exploration by @asiermartinez 

<img width="694" alt="157049274-d5f145bb-52cc-4b17-88f3-e596ed6b12db" src="https://user-images.githubusercontent.com/980622/157862362-9bd344c7-f90b-49d4-8ffa-af8520e2c648.png">

### Consider
- Selects with multi selections
- Selects with a filter input

## Links
[Slack Thread (external)](https://github.slack.com/archives/CSGAVNZ19/p1646964923715169)
[SelectMenu](https://primer.style/css/components/select-menu#filter)
[SelectPanel](https://primer.style/react/SelectPanel)

**Comment by asiermartinez** [comment]
@maximedegreve Leaving here the original Figma with all the _noodles_ in case anyone finds it useful.

https://www.figma.com/file/aTwOaUI8qt1QXiMmpta32C/DSP-component-library?node-id=1147%3A57452

**Comment by emilybrick** [comment]
This issue seems a bit out of date - we need to add SelectPanel guidelines to the new docs altogether when we move back to the v1 component. Taking over this issue for now @maximedegreve give me a shout if you want me to open a new one

**Comment by emilybrick** [comment]
Closing for now - this feature will not be included in the initial rollout of the SelectPanel improvements for FY25 q3/q4
